src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -235,7 +236,8 @@ CustomAttributeArgument[] ReadCustomAttributeArguments (XPathNavigator nav, Type
 			case MetadataType.Object:
 				var argumentIterator = nav.SelectChildren ("argument", string.Empty);
 				if (argumentIterator?.MoveNext () != true) {
-					_context.LogError ($"Custom attribute argument for 'System.Object' requires nested 'argument' node.", 1043);
+					_context.LogError (new DiagnosticString (DiagnosticId.CustomAttributeArgumentForTypeRequiresNestedNode).GetMessage ("System.Object", "argument"),
```

Comment:
This looks quite ugly, can you add overload which can simplify the code to something like

```c#
_context.LogError (DiagnosticId.CustomAttributeArgumentForTypeRequiresNestedNode, "System.Object", "argument");
```

Reply:
I couldn't implement exactly this behavior but I implemented something very similar. PTAL